httpbp/server.go

```diff
@@ -30,11 +43,12 @@ type httpHandlerFactory struct {
 	middlewares []Middleware
 }
 
-func (f httpHandlerFactory) NewHandler(name string, handle HandlerFunc, middlewares ...Middleware) http.Handler {
-	wrappers := make([]Middleware, 0, len(f.middlewares)+len(middlewares))
+func (f httpHandlerFactory) NewHandler(endpoint Endpoint) http.Handler {
+	wrappers := make([]Middleware, 0, len(f.middlewares)+len(endpoint.Middlewares)+1)
+	wrappers = append(wrappers, SupportedMethods(endpoint.Methods[0], endpoint.Methods[1:]...))
```

Comment:
🔕 til this won't cause panic when the slice contains one item: https://play.golang.org/p/HtkVu8laHF_n